identity/tests/rpc_endpoints.rs

```diff
@@ -0,0 +1,515 @@
+    Pool::new(ConnectionManager::new(connection_url)).unwrap()
+}
+
+pub async fn identity_client(
+    addr: SocketAddr,
+) -> std::io::Result<identity::rpc::service::IdentityServiceClient> {
+    identity::rpc::service::IdentityServiceClient::new(
+        Config::default(),
+        tarpc::serde_transport::tcp::connect(addr, Json::default).await?,
+    )
+    .spawn()
+}
+
+// get a valid user
+#[tokio::test]
+async fn get_user_exists() {
+    let configuration = Arc::new(get_test_configuration());
+    let db_test_context = DbTestContext::new(
+        configuration.db_connection_base_url(),
+        stdext::function_name!().into(),
+    );
+    let db: Arc<Db> =
+        Arc::new(Pool::new(ConnectionManager::new(db_test_context.get_connection_url())).unwrap());
+    let (server, socket) = rpc_server(
+        configuration.rpc_socket(),
+        configuration.clone(),
+        db.clone(),
+    )
+    .await
+    .unwrap();
+    tokio::spawn(server);
+    let mut client = identity_client(socket).await.unwrap();
+
+    let result = client.get_user(context::current(), 2).await.unwrap();
+
+    let expected_result = identity::rpc::service::User {
+        id: 2,
+        sub: "2".into(),
+        email: "jack.kerr@example.net".into(),
+        given_name: "Jack".into(),
+        family_name: "Kerr".into(),
+        picture: "https://example.com/avatar.jpg".into(),
+        active: true,
+    };
+
+    assert_eq!(Ok(expected_result), result);
+}
+
+// get an invalid user
+#[tokio::test]
+async fn get_user_not_exists() {
+    let configuration = Arc::new(get_test_configuration());
+    let db_test_context = DbTestContext::new(
+        configuration.db_connection_base_url(),
+        stdext::function_name!().into(),
+    );
+    let db: Arc<Db> =
+        Arc::new(Pool::new(ConnectionManager::new(db_test_context.get_connection_url())).unwrap());
+    let (server, socket) = rpc_server(
+        configuration.rpc_socket(),
+        configuration.clone(),
+        db.clone(),
+    )
+    .await
+    .unwrap();
+    tokio::spawn(server);
+    let mut client = identity_client(socket).await.unwrap();
+
+    let result = client.get_user(context::current(), 20).await.unwrap();
+
+    assert_eq!(Err(helpers::rpc::Error::NotFound), result);
+}
+
+// list users with offset/limit
+#[tokio::test]
+async fn list_users_exists() {
+    let configuration = Arc::new(get_test_configuration());
+    let db_test_context = DbTestContext::new(
+        configuration.db_connection_base_url(),
+        stdext::function_name!().into(),
+    );
+    let db: Arc<Db> =
+        Arc::new(Pool::new(ConnectionManager::new(db_test_context.get_connection_url())).unwrap());
+    let (server, socket) = rpc_server(
+        configuration.rpc_socket(),
+        configuration.clone(),
+        db.clone(),
+    )
+    .await
+    .unwrap();
+    tokio::spawn(server);
+    let mut client = identity_client(socket).await.unwrap();
```

Comment:
This seems to be at the beginning of every function...

Reply:
...and will be cleaned up. :-D